### PR TITLE
feat: Progress Bar Additions

### PR DESCRIPTION
**Pull request by Konargus**
## Purpose of this PR
- UnityRequest.downloadProgress was always 0 or 100.
- BatchMode Debug.Log(progress)

### Manual testing status
Tested 🧪 

## Comments to reviewers
On Merge/Release, please update Roomful Repo and poke me to implement it there.
Thanks.

### CODE REVIEW

File: Editor/EditorUtilities/EditorWebRequest.cs [hunk 1]
```diff
@@ -9,9 +10,33 @@ namespace StansAssets.Foundation.Editor
     /// </summary>
     public class EditorWebRequest
     {
+        /// <summary>
+        /// UnityWebRequest handler type.
+        /// </summary>
+        public enum HandlerType
+        {
+            ///<summary>Used to download data</summary>
```
Comment (stan-osipov) [inline Editor/EditorUtilities/EditorWebRequest.cs:18]:
```suggestion
            ///<summary>Used to download data.</summary>

```

File: Editor/EditorUtilities/EditorWebRequest.cs [hunk 2]
```diff
@@ -9,9 +10,33 @@ namespace StansAssets.Foundation.Editor
...
     public class EditorWebRequest
     {
+        /// <summary>
+        /// UnityWebRequest handler type.
+        /// </summary>
+        public enum HandlerType
+        {
+            ///<summary>Used to download data</summary>
+            Download,
+            ///<summary>Used to upload data</summary>
```
Comment (stan-osipov) [inline Editor/EditorUtilities/EditorWebRequest.cs:20]:
```suggestion
            ///<summary>Used to upload data.</summary>

```

File: Editor/EditorUtilities/EditorWebRequest.cs [hunk 3]
```diff
@@ -40,9 +65,10 @@ public EditorWebRequest(UnityWebRequest request)
         /// Request will display editor progress dialog with the given title.
         /// </summary>
         /// <param name="title">Editor progress dialog title.</param>
-        public void AddEditorProgressDialog(string title)
+        public void AddEditorProgressDialog(string title, HandlerType handlerType = HandlerType.Download)
```
Comment (stan-osipov) [inline Editor/EditorUtilities/EditorWebRequest.cs:68]: add doc for `handlerType `

File: Editor/EditorUtilities/EditorWebRequest.cs [hunk 4]
```diff
@@ -67,12 +93,24 @@ public void Send(Action<UnityWebRequest> callback)
...
         {
             if (m_ShowProgress)
             {
-                var progress = $"Download Progress: {Convert.ToInt32(UnityRequest.downloadProgress * 100f)}%";
-                EditorUtility.DisplayProgressBar(m_ProgressDialogTitle, progress, UnityRequest.downloadProgress);
+                if (!Application.isBatchMode)
+                {
+                    if (Time.realtimeSinceStartup > m_LastUpdate + 1)
+                    {
+                        Debug.Log(HandlerProgress);
```
Comment (stan-osipov) [inline Editor/EditorUtilities/EditorWebRequest.cs:106]: I don't think this is a good idea to spam console. 
If you need this - maybe we can have kind of a "debug mode".

**Review by stan-osipov — CHANGES_REQUESTED** [review]
minor changes requested.

File: Editor/EditorUtilities/EditorWebRequest.cs (hunk 4 above)
Comment (Konargus) [reply Editor/EditorUtilities/EditorWebRequest.cs:106]: In batchmode we don't have anything except console. If file is large, for ex. 1Gb, and network is slow we will never know if something is happening.

**Review by stan-osipov — APPROVED** [review]
thx!

**Pull request merged**